spring-ai-core/src/main/java/org/springframework/ai/model/function/FunctionCallbackContext.java

```diff
@@ -109,15 +114,23 @@ public FunctionCallback getFunctionCallback(@NonNull String beanName, @Nullable
 					.schemaType(this.schemaType)
 					.description(functionDescription)
 					.function(beanName, KotlinDelegate.wrapKotlinFunction(bean))
-					.inputType(functionInputClass)
+					.inputType(ParameterizedTypeReference.forType(functionInputType.getType()))
 					.build();
 			}
 			else if (KotlinDelegate.isKotlinBiFunction(functionType.toClass())) {
 				return FunctionCallback.builder()
 					.description(functionDescription)
 					.schemaType(this.schemaType)
 					.function(beanName, KotlinDelegate.wrapKotlinBiFunction(bean))
-					.inputType(functionInputClass)
+					.inputType(ParameterizedTypeReference.forType(functionInputType.getType()))
+					.build();
+			}
+			else if (KotlinDelegate.isKotlinSupplier(functionType.toClass())) {
+				return FunctionCallback.builder()
+					.description(functionDescription)
+					.schemaType(this.schemaType)
+					.function(beanName, KotlinDelegate.wrapKotlinSupplier(bean))
```

Comment:
Maybe we will need support for the Kotlin's equivalent of java.util.Consumer? 
But this can be implemented in a follow up PR.